### PR TITLE
Add Support for Vercel DNS

### PR DESCRIPTION
**Pull request by denkristoffer**
Docs: https://vercel.com/docs/api#endpoints/dns

I'm getting some weird output at the end of running the script with `--debug 2`, not sure if I made a mistake somewhere or if it's expected?

> _on_issue_success
> '' does not contain 'dns'

This PR closes #2087 – Zeit has rebranded to Vercel recently.

### CODE REVIEW

**Comment by Neilpang** [comment]
did you check this guide ?
https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Dev-Guide

**Comment by denkristoffer** [comment]
Yes, although I see I forgot to remove a few `_debug` calls. I've updated now.

**Comment by Neilpang** [comment]
please rebase code and fix all CI checks.

**Comment by cravend** [comment]
This worked for me, but I'm still getting the warning:
```
[Sat Aug 28 20:13:38 CEST 2021] _on_issue_success
[Sat Aug 28 20:13:38 CEST 2021] '' does not contain 'dns'
```

**Comment by denkristoffer** [comment]
@Neilpang Rebased, please allow the workflows to run 🙂

**Comment by Neilpang** [comment]
fix the dns api checks:
https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Test

**Comment by cravend** [comment]
Any status updates on this?

**Comment by Neilpang** [comment]
The CI actions are not passing yet.

https://github.com/denkristoffer/acme.sh/actions/runs/1212270375

**Comment by denkristoffer** [comment]
Finally looks green: https://github.com/denkristoffer/acme.sh/actions/runs/2284854723

Ready to merge? 🙂

File: dnsapi/dns_vercel.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  fi
+
+  _vercel_rest GET "v2/domains/$_domain/records"
+
+  count=$(printf "%s\n" "$response" | _egrep_o "\"name\":\"$_sub_domain\",[^{]*\"type\":\"TXT\"" | wc -l | tr -d " ")
+
+  if [ "$count" = "0" ]; then
+    _info "Don't need to remove."
+  else
+    _record_id=$(printf "%s" "$response" | _egrep_o "\"id\":[^,]*,\"slug\":\"[^,]*\",\"name\":\"$_sub_domain\",[^{]*\"type\":\"TXT\"" | cut -d: -f2 | cut -d, -f1 | tr -d '"')
```
Comment (Neilpang) [inline dnsapi/dns_vercel.sh:66]: There could be multiple  records with the same sub domain but different txt values;

for example:

```
_acme-challenge.example.com    TXT       TXTValueAAAAAAAA
_acme-challenge.example.com    TXT       TXTValueBBBBBBBBB
```


You should not remove all of them here.  you should only remove the one that you added in the "add()" method.

**Pull request merged**